### ISSUE TITLE
Test Accuracy is vastly different when using one vs multi core tpu implementations

### ISSUE THREAD

**Issue by KaleabTessera** [pytorch/xla#2565]
## 🐛 Bug

The test accuracy when using a single vs multi-core tpu setup is vastly different. This is with using `mesh_reduce` for the multi-core use case. 

Here are the test accuracy differences of resnet-18 on cifar-10:
- End of 10 epochs:     Multi Core - 61.11%           Single Core -   81.38%
- End of 20 epochs:     Multi Core - 79.57%           Single Core -   86.40%
- End of 50 epochs:     Multi Core - 85.42%           Single Core -   89.60%

## To Reproduce
I have a created a [colab notebook](https://colab.research.google.com/drive/1-oi2I7K4cacMskBa9tWu5mwHgcf6I0h1?usp=sharing) based on the [resnet-18 xla training example](https://github.com/pytorch/xla/blob/master/contrib/colab/resnet18-training.ipynb).

1. Run all cells in colab notebook.
2. See the vast performance difference in multi-core vs single-core implementations. 

## Expected behavior

Similiar performance levels. 

## Environment

 - Reproducible on XLA backend [CPU/TPU]: Google Colab TPU backed.
 - torch_xla version: nightly

**Comment by jysohn23** [comment]
You're changing the global batch size, and although the learning rate is also being tuned accordingly it's expected that larger global bsz wont have same loss curve. So I'd say this is working as intended and not specific to PyTorch/XLA. If you'd like, you can try set the same global batch size on multi core (if single core is bsz=8 just set per core bsz on multicore to be bsz=1 and use same LR) and you should see same loss/accuracy curves.

**Comment by KaleabTessera** [comment]
Ah I see that makes sense, thanks @jysohn23 . I will try that later when I have access to tpu on colab again.

Are there any best practices when setting TPU batch size? I want to get close to SOTA for some deep models (wideresnet, resnext etc), but those batch sizes are configured for a single gpu.

**Comment by jysohn23** [comment]
It's hard to say since it's just another hyperparameter, which depends on your model, dataset, etc. For hardware utilization's sake, it's usually best to maximize batch size as long as it can fit in the chip memory, but you'll have to tweak hparams accordingly.

**Comment by taylanbil** [comment]
One rule of thumb is, with global batch size large, each gradient has more "signal" than "noise". Therefore, taking a larger step could be a good idea; i.e. w/ large global batch size, larger learning rates can be useful and lead to faster convergence.

**Comment by KaleabTessera** [comment]
Great, thanks so much for the input @jysohn23  @taylanbil , much appreciated.
